user/docker.md

```diff
@@ -24,6 +24,8 @@ examples.
 
 > Travis CI automatically routes builds to run on Trusty `sudo: required` when `services: docker` is configured.
 > We do not currently support use of Docker on OS X.
+> This feature is only available on travis-ci.com and travis-ci.org
```

Comment:
Hey @schultyy 

Great idea to clear this up, but I think we need to change the wording. That to me reads like this feature isn't available, here is how to do it :-p

How about just 

> For information on how to use Docker on Travis CI Enterprise check out [ ] ...

Reply:
Sounds good - yeah the words I picked weren't really the best here 😅

Reply:
It is a pretty natural consequence of adding the Enterprise stuff on to the rest of the docs. They weren't written with that in mind, so there is often no natural connection.